### PR TITLE
MODPUBSUB-46 Created Consumer Service

### PR DESCRIPTION
**Pull request by KaterynaSenchenko**

### CODE REVIEW

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        return Future.succeededFuture();
+      });
+  }
+
+  private Handler<AsyncResult<HttpClientResponse>> getEventDeliveredHandler(Event event, String tenantId, MessagingModule subscriber) {
+    return ar -> {
+      if (ar.failed()) {
+        LOGGER.error("Event {} was not delivered to {}", ar.cause(), event.getId(), subscriber.getSubscriberCallback());
+        saveAuditMessage(event, tenantId, AuditMessage.State.REJECTED);
+      } else if (ar.result().statusCode() != 200 && ar.result().statusCode() != 201 && ar.result().statusCode() != 204) {
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: HttpStatus constants plz

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      .withTenantId(tenantId)
+      .withModuleRole(SUBSCRIBER)
+      .withEventType(event.getEventType()))
+      .compose(messagingModuleList -> {
+        if (CollectionUtils.isEmpty(messagingModuleList)) {
+          String errorMessage = format("There is no SUBSCRIBERS registered for event type %s. Event %s will not be delivered", event.getEventType(), event.getId());
+          LOGGER.error(errorMessage);
+        } else {
+          messagingModuleList
+            .forEach(subscriber -> doRequest(event, subscriber.getSubscriberCallback(), okapiHeaders)
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: plz use parallel stream

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          event.getId(), subscriber.getSubscriberCallback(), ar.result().statusCode(), ar.result().statusMessage());
+        saveAuditMessage(event, tenantId, AuditMessage.State.REJECTED);
+      } else {
+        LOGGER.info("Delivered event {} to {}", event.getId(), subscriber.getSubscriberCallback());
+        saveAuditMessage(event, tenantId, AuditMessage.State.DELIVERED);
+      }
+    };
+  }
+
+  private Future<HttpClientResponse> doRequest(Event event, String callbackPath, Map<String, String> okapiHeaders) {
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: Do we have such method at RestUtils?

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  /**
+   * Creates a consumer and subscribes it to particular topics
+   *
+   * @param moduleId     module id
+   * @param eventTypes   list of event types that specified module is subscribing to receive
+   * @param tenantId     tenant id
+   * @param okapiHeaders okapi headers
+   * @return future with true if succeeded
+   */
+  Future<Boolean> subscribe(String moduleId, List<String> eventTypes, String tenantId, Map<String, String> okapiHeaders);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java]: String tenantId, Map<String, String> okapiHeaders plz wrap into the object like OkapiConnectionParams

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java (hunk 3 above)
Comment (KaterynaSenchenko) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: We do not have data-import-utils dependency in pub-sub. This method is based on that implementation, but adjusted/simplified a little for pub-sub needs

File: mod-pubsub-server/src/test/java/org/folio/services/impl/ConsumerServiceUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      .withId(UUID.randomUUID().toString())
+      .withEventType(EVENT_TYPE)
+      .withEventMetadata(new EventMetadata()
+        .withTenantId(TENANT)
+        .withEventTTL(30)
+        .withPublishedBy("mod-very-important-1.0.0"));
+
+    OkapiConnectionParams params = new OkapiConnectionParams();
+    params.setHeaders(headers);
+    params.setOkapiUrl(headers.getOrDefault("x-okapi-url", "localhost"));
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/test/java/org/folio/services/impl/ConsumerServiceUnitTest.java]: use final constants

**Pull request merged**